nautobot/docs/development/best-practices.md

```diff
@@ -63,3 +63,193 @@ class ExampleModel(PrimaryModel):
+
+### Filter Naming and Definition
+
+- Boolean filters for membership **must** be named with `has_{related_name}` (e.g. `has_interfaces`)
+
+- Boolean filters for identity **must** be named with `is_{name}` (e.g. `is_virtual_chassis`) although this is semantically identical to `has_` filters, there may be occasions where naming the filter `is_` would be more intuitive.
+
+- Filters **must** declare [`field_name`](https://django-filter.readthedocs.io/en/stable/ref/filters.html#field-name) when they have a different name than the underlying field they are referencing. Where possible the suffix component of the filter name should map directly to the underlying field name.
+
+  For example, `DeviceFilterSet.has_console_ports` could be better named, to assert that the filter name following the `has_` prefix is a one-to-one mapping to the underlying model's related field name (`consoleports`) which also means that `field_name` is not required:
+
+```python
+    has_consoleports = BooleanFilter(...)
+```
+
+- Filters **must** be declared using the appropriate lookup expression (`lookup_expr`) if any other expression than `exact` (the default) is required. For example:
+
+```python
+   has_consoleports = BooleanFilter(lookup_expr="isnull", ...)
+```
+
+- Filters **must** be declared using [`exclude=True`](https://django-filter.readthedocs.io/en/stable/ref/filters.html#exclude) if a queryset `.exclude()` is required to be called vs. queryset `.filter()` which is the default when the filter default `exclude=False` is passed through. If you require `Foo.objects.exclude()`, you must pass `exclude=True` instead of defining a filterset method to explicitly hard-code such a query. For example:
+
+```python
+   has_consoleports = BooleanFilter(lookup_expr="isnull", exclude=True)
+```
+
+- Filters **must** be declared using [`disinct=True`](https://django-filter.readthedocs.io/en/stable/ref/filters.html#distinct) if a queryset `.distinct()`is required to be called on the queryset
+- Filters **must not** be set to be required using `required=True`
+- Filter methods defined using the [`method=`](https://django-filter.readthedocs.io/en/stable/ref/filters.html#method) keyword argument **may only be used as a last resort** (see below) when correct usage of `lookup_expr` ,  `exclude`, or other filter keyword arguments do not suffice. In other words: filter methods should used as the exception and not the rule.
+- Use of [`filter_overrides`](https://django-filter.readthedocs.io/en/stable/ref/filterset.html#filter-overrides) **must be considered** in cases where local overrides to certain model field types are required vs. overloading or re-declaring the filter fields are required, such as changing a filter class, or customizing a UI widget. For example:
```

Comment:
I'm having a hard time parsing this sentence and understanding when it would apply. When would a "local override to certain model field types" be required? When would "overloading or re-declaring the filter fields" be required?

Reply:
Refactored.